ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -252,9 +254,53 @@ private[kvutils] class TransactionCommitter(
   private def blind: Step =
     (commitContext, transactionEntry) => {
       val blindingInfo = Blinding.blind(transactionEntry.transaction)
-      buildFinalResult(commitContext, transactionEntry, blindingInfo)
+      buildFinalResultState(commitContext, transactionEntry, blindingInfo)
     }
 
+
+
+  private[committer] def removeUnnecessaryNodes: Step = (commitContext, transactionEntry) => {
```

Comment:
Added a comment for this method

Reply:
Thanks, can you also expand it a bit to explain why we're doing that?

Reply:
@fabiotudone-da Sure, it is for reducing the burden on the flows downstream (i.e. we aim at increasing throughput in the reference architecture: see the ticket linked in the description). Although, just to make it clear, this is not going to be solved with this PR since it has breaking changes. This PR is just for allowing to see what the impact will be (from a performance POV) on the other components of the read path